.circleci/config.yml

```diff
@@ -29,35 +143,35 @@ workflows:
           requires:
             - fabric8_istio
 jobs:
+
+  build:
+    machine:
+      image: ubuntu-2204:2022.04.1
+    environment:
+      _JAVA_OPTIONS: "-Xms2g -Xmx2g"
+    steps:
+      - restore_main_cache
```

Comment:
there are _two_ caches here involved. the first one is the old `.m2` cache you are thinking of and we save it and restore it in the usual way. The second one is a cache with latest dependencies, and its key is `spring-cloud-kubernetes-per-run-based-cache-{{ .Environment.CIRCLE_WORKFLOW_ID }}` - thus unique per this run. When I need the cache to be restored in some step, I always restore this one, with latest snapshots. 

This is why that `ActuatorRefreshKafkaIT` was not failing in circleci - which was a problem/bug.